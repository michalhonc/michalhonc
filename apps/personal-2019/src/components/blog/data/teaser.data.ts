export const Query = {
  allMarkdownRemark: {
    edges: [
      {
        node: {
          frontmatter: {
            date: "2020-03-31",
            language: "en",
            description:
              "Canny tricks that won't make your React app better but will make your dev process easier.",
            path: "/blog/5-canny-tricks-for-react-devs",
            title: "5 Canny little tricks for React devs",
            category: "react",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png",
                  aspectRatio: 1.4184397163120568,
                  srcSet:
                    "/static/4e57a90ee8c2f36cfe0eba16def56413/69585/5_canny_react_tricks.png 200w,\n/static/4e57a90ee8c2f36cfe0eba16def56413/497c6/5_canny_react_tricks.png 400w,\n/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png 704w",
                  sizes: "(max-width: 704px) 100vw, 704px",
                },
              },
            },
          },
        },
      },
      {
        node: {
          frontmatter: {
            date: "2020-03-31",
            language: "cs",
            description:
              "Triky, které Vaši aplikaci nezlepší, ale usnadní vám proces vývoje",
            path: "/blog/5-triku-pro-vyvojare-reactu",
            title: "5 triků pro vývojáře Reactu",
            category: "react",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png",
                  aspectRatio: 1.4184397163120568,
                  srcSet:
                    "/static/4e57a90ee8c2f36cfe0eba16def56413/69585/5_canny_react_tricks.png 200w,\n/static/4e57a90ee8c2f36cfe0eba16def56413/497c6/5_canny_react_tricks.png 400w,\n/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png 704w",
                  sizes: "(max-width: 704px) 100vw, 704px",
                },
              },
            },
          },
        },
      },
      {
        node: {
          frontmatter: {
            date: "2020-02-25",
            language: "cs",
            description:
              "Nepřicházejte kvůli AdBlocku o peníze z reklamy! Zjištěte 7 způsobů jak se proti AdBlocku můžete bránit.",
            path: "/blog/7-zpusobu-jak-obejit-adblock",
            title: "7 způsobů jak obejít AdBlock",
            category: "reklama",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png",
                  aspectRatio: 1.492537313432836,
                  srcSet:
                    "/static/d96fb15f308d10e6d5035097232fbd35/69585/adblock_thumbnail.png 200w,\n/static/d96fb15f308d10e6d5035097232fbd35/497c6/adblock_thumbnail.png 400w,\n/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png 500w",
                  sizes: "(max-width: 500px) 100vw, 500px",
                },
              },
            },
          },
        },
      },
      {
        node: {
          frontmatter: {
            date: "2020-02-25",
            language: "en",
            description:
              "Don't lose advertising money because of AdBlock! Find out 7 ways you can defend yourself against AdBlock.",
            path: "/blog/7-ways-to-bypass-adblock",
            title: "7 ways to bypass AdBlock",
            category: "ads",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png",
                  aspectRatio: 1.492537313432836,
                  srcSet:
                    "/static/d96fb15f308d10e6d5035097232fbd35/69585/adblock_thumbnail.png 200w,\n/static/d96fb15f308d10e6d5035097232fbd35/497c6/adblock_thumbnail.png 400w,\n/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png 500w",
                  sizes: "(max-width: 500px) 100vw, 500px",
                },
              },
            },
          },
        },
      },
      {
        node: {
          frontmatter: {
            date: "2020-04-08",
            language: "en",
            description:
              "In this tutorial I'll show you how to set up professional development process with help of continuous deployment.",
            path: "/blog/continuous-deployment-for-free",
            title:
              "Continuous development for FREE with Github, Travis CI and Heroku",
            category: "continuous deployment",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png",
                  aspectRatio: 1.3605442176870748,
                  srcSet:
                    "/static/f53b530a16d87556855e6e551c169f83/69585/travis_thumbnail.png 200w,\n/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png 387w",
                  sizes: "(max-width: 387px) 100vw, 387px",
                },
              },
            },
          },
        },
      },
      {
        node: {
          frontmatter: {
            date: "2020-04-08",
            language: "cs",
            description:
              "V tomto tutorialu vám ukáži, jak nastavit profesionální vývojové prostřední s pomocí continuous deployment.",
            path: "/blog/continuous-deployment-zdarma",
            title: "Continuous development zdarma s Github, Travis CI a Heroku",
            category: "continuous deployment",
            thumbnail: {
              childImageSharp: {
                fluid: {
                  src: "/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png",
                  aspectRatio: 1.3605442176870748,
                  srcSet:
                    "/static/f53b530a16d87556855e6e551c169f83/69585/travis_thumbnail.png 200w,\n/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png 387w",
                  sizes: "(max-width: 387px) 100vw, 387px",
                },
              },
            },
          },
        },
      },
    ],
  },
};
