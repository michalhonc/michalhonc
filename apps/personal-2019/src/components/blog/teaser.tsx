import * as React from "react";
import { useStaticQuery, graphql } from "gatsby";
import { useIntl, Link, WrappedComponentProps } from "gatsby-plugin-intl";
import Img from "gatsby-image/withIEPolyfill";

import { prepareData } from "../../pages/blog";

const Query = graphql`
  {
    allMarkdownRemark {
      edges {
        node {
          frontmatter {
            date
            language
            title
            path
            description
            thumbnail {
              childImageSharp {
                fluid {
                  ...GatsbyImageSharpFluid_withWebp
                }
              }
            }
          }
        }
      }
    }
  }
`;

const MorePosts = ({ intl }: WrappedComponentProps) => (
  <>
    {intl.formatMessage({ id: "more_posts" })}
    {String.fromCharCode(160)}
    {String.fromCharCode(187)}
  </>
);

export const Teaser: React.FC = () => {
  const intl = useIntl();
  const { allMarkdownRemark } = useStaticQuery(Query);
  const [firstPost, ...others] = prepareData(allMarkdownRemark, intl);

  return (
    <section className="Container Teaser">
      <h2
        dangerouslySetInnerHTML={{
          __html: intl.formatMessage({ id: "teaser_title" }),
        }}
      />

      <div className="Teaser-wrapper">
        {others.slice(0, 2).map(({ node }) => {
          const { title, path, description, thumbnail } = node.frontmatter;
          return (
            <article key={path}>
              <Link to={path} aria-label={`Link to blog post: ${title}`}>
                <Img
                  objectFit="contain"
                  fluid={thumbnail.childImageSharp.fluid}
                />
              </Link>
              <div className="Teaser-row">
                <h3>{title}</h3>
                <p>{description}</p>
                <Link className="underline" to={path}>
                  <MorePosts intl={intl} />
                </Link>
              </div>
            </article>
          );
        })}

        {firstPost && (
          <article key={firstPost.node.frontmatter.path}>
            <Link
              to={firstPost.node.frontmatter.path}
              aria-label={`Link to blog post: ${firstPost.node.frontmatter.title}`}
            >
              <Img
                objectFit="contain"
                fluid={
                  firstPost.node.frontmatter.thumbnail.childImageSharp.fluid
                }
              />
            </Link>
            <div className="Teaser-row">
              <h3>{firstPost.node.frontmatter.title}</h3>
              <p>{firstPost.node.frontmatter.description}</p>
              <Link className="underline" to={firstPost.node.frontmatter.path}>
                <MorePosts intl={intl} />
              </Link>
            </div>
          </article>
        )}
      </div>

      <Link className="Teaser-more underline" to="/blog">
        <MorePosts intl={intl} />
      </Link>
    </section>
  );
};
