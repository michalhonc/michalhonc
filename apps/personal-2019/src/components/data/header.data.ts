export const HeaderQuery = {
  background: {
    childImageSharp: {
      fluid: {
        src: "/static/45779372c1991322e2cf8821eb9b133f/74fd5/header-background.jpg",
        aspectRatio: 1.3333333333333333,
        srcSet:
          "/static/45779372c1991322e2cf8821eb9b133f/e3903/header-background.jpg 480w,\n/static/45779372c1991322e2cf8821eb9b133f/b31d1/header-background.jpg 960w,\n/static/45779372c1991322e2cf8821eb9b133f/74fd5/header-background.jpg 1920w,\n/static/45779372c1991322e2cf8821eb9b133f/cb67e/header-background.jpg 2483w",
        sizes: "(max-width: 1920px) 100vw, 1920px",
      },
    },
  },
  foreground: {
    childImageSharp: {
      fluid: {
        src: "/static/96b4dfdae8e006ed6b85bfb9aadd902b/0e6e2/header-foreground.png",
        aspectRatio: 1.3333333333333333,
        srcSet:
          "/static/96b4dfdae8e006ed6b85bfb9aadd902b/b460a/header-foreground.png 480w,\n/static/96b4dfdae8e006ed6b85bfb9aadd902b/6e898/header-foreground.png 960w,\n/static/96b4dfdae8e006ed6b85bfb9aadd902b/0e6e2/header-foreground.png 1920w,\n/static/96b4dfdae8e006ed6b85bfb9aadd902b/4e88d/header-foreground.png 2880w,\n/static/96b4dfdae8e006ed6b85bfb9aadd902b/049dd/header-foreground.png 3309w",
        sizes: "(max-width: 1920px) 100vw, 1920px",
      },
    },
  },
};
