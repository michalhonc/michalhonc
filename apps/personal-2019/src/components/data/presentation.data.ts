export const PresentationQuery = {
  mockup: {
    childImageSharp: {
      fluid: {
        aspectRatio: 0.7092198581560284,
        src: "/static/ea55a42170bb63d820b1b81f9df1bf59/3891b/mockup.png",
        srcSet:
          "/static/ea55a42170bb63d820b1b81f9df1bf59/3ba9f/mockup.png 100w,\n/static/ea55a42170bb63d820b1b81f9df1bf59/8ac63/mockup.png 200w,\n/static/ea55a42170bb63d820b1b81f9df1bf59/3891b/mockup.png 400w,\n/static/ea55a42170bb63d820b1b81f9df1bf59/8c332/mockup.png 600w,\n/static/ea55a42170bb63d820b1b81f9df1bf59/bc8e0/mockup.png 800w,\n/static/ea55a42170bb63d820b1b81f9df1bf59/8d364/mockup.png 1500w",
        sizes: "(max-width: 400px) 100vw, 400px",
      },
    },
  },
};
