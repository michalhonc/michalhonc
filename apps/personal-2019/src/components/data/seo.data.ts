export const SeoQuery = {
  site: {
    siteMetadata: {
      author: "Michal Honc",
      description:
        "První konzultace ZDARMA! Vytvořím Vám rychlé a moderní webové stránky, se kterýma Vy nebo Váš produkt uspějete. Sjednejte si konzultaci zdarma během jedné minuty",
      title: "Michal Honc",
    },
  },
};
