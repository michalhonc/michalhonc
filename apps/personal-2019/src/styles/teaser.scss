.Teaser {
  height: 55rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 12rem auto 18rem;
  padding: 0;

  @include respond('tab-port') {
    margin: 12rem auto 12rem;
  }

  &-wrapper {
    display: flex;
    flex-flow: column wrap;
    height: 100%;
    margin: 3rem 0 0 0;

    @include respond('tab-port') {
      flex-flow: column-reverse;
      margin: 2rem 1rem;
    }
  }

  &-more {
    font-size: 2.2rem;
    font-weight: bold;
    color: var(--c-beta);

    @include respond('tab-port') {
      font-size: 2rem;
      padding: 0 1rem;
    }

    &:hover {
      text-decoration: underline;
    }
  }

  & article {
    display: flex;
    flex-direction: column;
    width: 60%;
    flex: 1;
    height: 50%;
    padding: 2rem;

    @include respond('tab-port') {
      width: 100%;
      height: auto;
      padding: 0.2rem;
    }

    & a:hover {
      text-decoration: underline;
    }

    &:not(:last-of-type) {
      flex-direction: row;

      & .gatsby-image-wrapper {
        width: 20rem;
        flex-shrink: 0;
        margin-right: 2rem;

        @include respond('tab-port') {
          width: 10rem;
          margin-right: 1rem;
        }
      }

      & > .Teaser-row {
        display: flex;
        flex-direction: column;
        justify-content: space-evenly;
        align-items: flex-start;
      }
    }

    &:last-of-type {
      height: 100%;
      width: 40%;

      @include respond('tab-port') {
        flex-direction: row;
        width: 100%;
      }


      & .gatsby-image-wrapper {
        max-height: 17rem;
        @include respond('tab-port') {
          width: 10rem;
          margin-right: 1rem;
        }
      }

      & > .Teaser-row {
        display: flex;
        flex-direction: column;
        justify-content: space-evenly;
        align-items: flex-start;
      }
    }

    & a,
    & p {
      font-size: 2rem;
      color: var(--c-beta);

      @include respond('tab-port') {
        font-size: 1.6rem;
      }
    }

    & a {
      color: var(--c-black);
    }

  }
}
