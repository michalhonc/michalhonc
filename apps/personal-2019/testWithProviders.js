import React from "react";
import renderer from "react-test-renderer";
import { IntlContextProvider } from "gatsby-plugin-intl";

export const renderWithProviders = (component) =>
  renderer.create(
    <IntlContextProvider value={{ language: "en-US", routed: true }}>
      {component}
    </IntlContextProvider>
  );
