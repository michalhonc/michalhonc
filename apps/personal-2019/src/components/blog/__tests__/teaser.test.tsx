import * as React from "react";
import * as Gatsby from "gatsby";
import renderer from "react-test-renderer";
import { renderWithProviders } from "../../../../testWithProviders";
import { Teaser } from "../teaser";
import { Query } from "../data/teaser.data";

describe("teaser", () => {
  const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery");
  useStaticQuery.mockImplementation(() => Query);

  it("renders correctly", () => {
    const tree = renderWithProviders(<Teaser />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
