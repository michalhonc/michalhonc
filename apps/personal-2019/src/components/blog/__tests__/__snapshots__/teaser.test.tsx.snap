// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teaser renders correctly 1`] = `
<section
  className="Container Teaser"
>
  <h2
    dangerouslySetInnerHTML={
      Object {
        "__html": "Nejnovější články na <span class='underline'>blogu</span>",
      }
    }
  />
  <div
    className="Teaser-wrapper"
  >
    <article>
      <a
        aria-label="Link to blog post: 5 triků pro vývojáře Reactu"
        href="/en-US/blog/5-triku-pro-vyvojare-reactu"
        onClick={[Function]}
      >
        <div
          className=" gatsby-image-wrapper"
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            aria-hidden={true}
            style={
              Object {
                "paddingBottom": "70.5%",
                "width": "100%",
              }
            }
          />
          <picture>
            <source
              sizes="(max-width: 704px) 100vw, 704px"
              srcSet="/static/4e57a90ee8c2f36cfe0eba16def56413/69585/5_canny_react_tricks.png 200w,
/static/4e57a90ee8c2f36cfe0eba16def56413/497c6/5_canny_react_tricks.png 400w,
/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png 704w"
            />
            <img
              alt=""
              loading="lazy"
              onLoad={[Function]}
              sizes="(max-width: 704px) 100vw, 704px"
              src="/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png"
              srcSet="/static/4e57a90ee8c2f36cfe0eba16def56413/69585/5_canny_react_tricks.png 200w,
/static/4e57a90ee8c2f36cfe0eba16def56413/497c6/5_canny_react_tricks.png 400w,
/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png 704w"
              style={
                Object {
                  "fontFamily": "\\"object-fit: contain; object-position: 50% 50%\\"",
                  "height": "100%",
                  "left": 0,
                  "objectFit": "contain",
                  "objectPosition": "50% 50%",
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transition": "opacity 500ms",
                  "width": "100%",
                }
              }
            />
          </picture>
          <noscript
            dangerouslySetInnerHTML={
              Object {
                "__html": "<picture><source srcset=\\"/static/4e57a90ee8c2f36cfe0eba16def56413/69585/5_canny_react_tricks.png 200w,
/static/4e57a90ee8c2f36cfe0eba16def56413/497c6/5_canny_react_tricks.png 400w,
/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png 704w\\" sizes=\\"(max-width: 704px) 100vw, 704px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 704px) 100vw, 704px\\" srcset=\\"/static/4e57a90ee8c2f36cfe0eba16def56413/69585/5_canny_react_tricks.png 200w,
/static/4e57a90ee8c2f36cfe0eba16def56413/497c6/5_canny_react_tricks.png 400w,
/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png 704w\\" src=\\"/static/4e57a90ee8c2f36cfe0eba16def56413/3ee52/5_canny_react_tricks.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
              }
            }
          />
        </div>
      </a>
      <div
        className="Teaser-row"
      >
        <h3>
          5 triků pro vývojáře Reactu
        </h3>
        <p>
          Triky, které Vaši aplikaci nezlepší, ale usnadní vám proces vývoje
        </p>
        <a
          className="underline"
          href="/en-US/blog/5-triku-pro-vyvojare-reactu"
          onClick={[Function]}
        >
          Více článků
           
          »
        </a>
      </div>
    </article>
    <article>
      <a
        aria-label="Link to blog post: 7 způsobů jak obejít AdBlock"
        href="/en-US/blog/7-zpusobu-jak-obejit-adblock"
        onClick={[Function]}
      >
        <div
          className=" gatsby-image-wrapper"
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            aria-hidden={true}
            style={
              Object {
                "paddingBottom": "67%",
                "width": "100%",
              }
            }
          />
          <picture>
            <source
              sizes="(max-width: 500px) 100vw, 500px"
              srcSet="/static/d96fb15f308d10e6d5035097232fbd35/69585/adblock_thumbnail.png 200w,
/static/d96fb15f308d10e6d5035097232fbd35/497c6/adblock_thumbnail.png 400w,
/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png 500w"
            />
            <img
              alt=""
              loading="lazy"
              onLoad={[Function]}
              sizes="(max-width: 500px) 100vw, 500px"
              src="/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png"
              srcSet="/static/d96fb15f308d10e6d5035097232fbd35/69585/adblock_thumbnail.png 200w,
/static/d96fb15f308d10e6d5035097232fbd35/497c6/adblock_thumbnail.png 400w,
/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png 500w"
              style={
                Object {
                  "fontFamily": "\\"object-fit: contain; object-position: 50% 50%\\"",
                  "height": "100%",
                  "left": 0,
                  "objectFit": "contain",
                  "objectPosition": "50% 50%",
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transition": "opacity 500ms",
                  "width": "100%",
                }
              }
            />
          </picture>
          <noscript
            dangerouslySetInnerHTML={
              Object {
                "__html": "<picture><source srcset=\\"/static/d96fb15f308d10e6d5035097232fbd35/69585/adblock_thumbnail.png 200w,
/static/d96fb15f308d10e6d5035097232fbd35/497c6/adblock_thumbnail.png 400w,
/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png 500w\\" sizes=\\"(max-width: 500px) 100vw, 500px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 500px) 100vw, 500px\\" srcset=\\"/static/d96fb15f308d10e6d5035097232fbd35/69585/adblock_thumbnail.png 200w,
/static/d96fb15f308d10e6d5035097232fbd35/497c6/adblock_thumbnail.png 400w,
/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png 500w\\" src=\\"/static/d96fb15f308d10e6d5035097232fbd35/46604/adblock_thumbnail.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
              }
            }
          />
        </div>
      </a>
      <div
        className="Teaser-row"
      >
        <h3>
          7 způsobů jak obejít AdBlock
        </h3>
        <p>
          Nepřicházejte kvůli AdBlocku o peníze z reklamy! Zjištěte 7 způsobů jak se proti AdBlocku můžete bránit.
        </p>
        <a
          className="underline"
          href="/en-US/blog/7-zpusobu-jak-obejit-adblock"
          onClick={[Function]}
        >
          Více článků
           
          »
        </a>
      </div>
    </article>
    <article>
      <a
        aria-label="Link to blog post: Continuous development zdarma s Github, Travis CI a Heroku"
        href="/en-US/blog/continuous-deployment-zdarma"
        onClick={[Function]}
      >
        <div
          className=" gatsby-image-wrapper"
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            aria-hidden={true}
            style={
              Object {
                "paddingBottom": "73.5%",
                "width": "100%",
              }
            }
          />
          <picture>
            <source
              sizes="(max-width: 387px) 100vw, 387px"
              srcSet="/static/f53b530a16d87556855e6e551c169f83/69585/travis_thumbnail.png 200w,
/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png 387w"
            />
            <img
              alt=""
              loading="lazy"
              onLoad={[Function]}
              sizes="(max-width: 387px) 100vw, 387px"
              src="/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png"
              srcSet="/static/f53b530a16d87556855e6e551c169f83/69585/travis_thumbnail.png 200w,
/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png 387w"
              style={
                Object {
                  "fontFamily": "\\"object-fit: contain; object-position: 50% 50%\\"",
                  "height": "100%",
                  "left": 0,
                  "objectFit": "contain",
                  "objectPosition": "50% 50%",
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transition": "opacity 500ms",
                  "width": "100%",
                }
              }
            />
          </picture>
          <noscript
            dangerouslySetInnerHTML={
              Object {
                "__html": "<picture><source srcset=\\"/static/f53b530a16d87556855e6e551c169f83/69585/travis_thumbnail.png 200w,
/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png 387w\\" sizes=\\"(max-width: 387px) 100vw, 387px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 387px) 100vw, 387px\\" srcset=\\"/static/f53b530a16d87556855e6e551c169f83/69585/travis_thumbnail.png 200w,
/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png 387w\\" src=\\"/static/f53b530a16d87556855e6e551c169f83/f0c16/travis_thumbnail.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
              }
            }
          />
        </div>
      </a>
      <div
        className="Teaser-row"
      >
        <h3>
          Continuous development zdarma s Github, Travis CI a Heroku
        </h3>
        <p>
          V tomto tutorialu vám ukáži, jak nastavit profesionální vývojové prostřední s pomocí continuous deployment.
        </p>
        <a
          className="underline"
          href="/en-US/blog/continuous-deployment-zdarma"
          onClick={[Function]}
        >
          Více článků
           
          »
        </a>
      </div>
    </article>
  </div>
  <a
    className="Teaser-more underline"
    href="/en-US/blog"
    onClick={[Function]}
  >
    Více článků
     
    »
  </a>
</section>
`;
